### DISCUSSION TITLE
[Authentication service] IDP Auth0

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#361] [Q&A] [answered]
### Which section?

Console

### Which is the version?

12

### Description

Hello guys,

I am currently implementing an ROPC flow using the authentication service and Auth0 as IDP following the provided documentation and I have a few questions.

This is my config file:
```
{
  "apps": {
    "web": {
      "providers": {
        "auth0": { 
          "type": "okta",
          "clientId": "hidden",
          "clientSecret": "hidden",
          "authUrl": "https://dev-hidden.us.auth0.com/authorize",
          "tokenUrl": "https://dev-hidden.us.auth0.com/oauth/token",
          "userInfoUrl": "https://dev-hidden.us.auth0.com/api/v2/users",
          "baseUrl": "https://dev-hidden.us.auth0.com",
          "scope": [
            "custom_scope"
          ],
          "order": 10
        }
      },
      "authorizeStateRequired": true,
      "redirectUrl": "hidden",
      "realm": "hidden",
      "isWebsiteApp": true,
      "defaultGroups": [
        "hidden"
      ],
      "issuer": "https://dev-hidden.us.auth0.com"
    }
  }
}
```

1. Is "okta" the correct type for Auth0 provider?
2. What is this error on the Authentication service referred to? (the baseUrl is actually an absolute url)
`{"level":60,"msg":"fails to create client: baseURL should be an absolute url","time":1703770220254}`

**Comment by fredmaggiowski** [comment, marked as answer]
Hi thanks for reaching out!

As of now there is no explicit support for auth0 but using the "generic" provider type should do the trick. The okta type is meant for integrating with the okta enterprise idp and it was implemented far before okta acquired auth0 so we can't guarantee it works (it may since the underneath flows are following the same specification but I would not recommend it)

As for ROPC support @malta895 can provide you with further information

**Reply by georgev-97** [reply]
Thank you, I will consider the two options

**Reply by malta895** [reply]
Hi @georgev-97,
for the ROPC usage in the authentication service you can refer to my answer on #360. Given the example configuration you provided, the `oauth/token` request body should look like the following:
```json
{
    "grant_type": "password",
    "username": "yourusername",
    "password": "yourpassword",
    "appId": "web",
    "providerId": "auth0"
}
```

You could also use the Auth0-client for this provider, if the authentication service does not fit your needs.

You can find all the details on how to configure it in the [Auth0 Client documentation](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/overview_and_usage).

In particular, for the ROPC flow, you can find an example cURL [here](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/overview_and_usage#how-to-enable-username-and-password-login).

**Reply by georgev-97** [reply]
This error `{"level":60,"msg":"fails to create client: baseURL should be an absolute url","time":1703770220254}` is caused by the wrong type in the provider?

**Reply by malta895** [reply]
That error might be related to the `USERS_CRUD_BASE_URL` env, could you check it is set correctly?
Make sure it is in the form `http://crud-service/users`

**Reply by georgev-97** [reply]
Thank you, that was the error!
